src/lxc/lsm/apparmor.c

```diff
@@ -159,61 +488,546 @@ static bool aa_needs_transition(char *curlabel)
 	return true;
 }
 
+static inline void uint64hex(char *buf, uint64_t num)
+{
+	size_t i;
+
+	buf[16] = 0;
+	for (i = 16; i--;) {
```

Comment:
Is this what you meant to do?  You'll cut out without handling i == 0 ?

Reply:
Even if you meant to do this, I think that
for (i = 16; i; i--)
would leave future readers less suspicious